pd/src/app.rs

```diff
@@ -471,7 +546,15 @@ impl Service<Request> for App {
                     .instrument(Span::current())
                     .boxed();
                 }
-                Request::EndBlock(end) => Response::EndBlock(self.end_block(end)),
+                Request::EndBlock(end) => {
+                    let rsp = self.end_block(end);
+                    return self
+                        .sequencer
+                        .execute(rsp)
+                        // TODO: (@hdevalence): why doesn't this span attach to the inner future?
+                        .instrument(Span::current())
+                        .boxed();
```

Comment:
The sequencer doesn't propagate spans correctly.

I think the issue is that the sequencer future is instrumenting the future that waits on the joinhandle, but not instrumenting the future that gets spawned as a new task.  I'm not sure what the best way to propagate the span is.  Here are some options:

- we could use `Span::current()` in `Sequencer::execute`, which grabs the ambient span; in this case, that's the right one, but it might not be elsewhere;
- we could have the caller instrument the future passed to `execute`, like `.execute(rsp.instrument(Span::current()))`; if we wanted to instrument the outer (returned) future, we'd need two `instrument` calls, but since it's just waiting on a `JoinHandle` maybe we don't care (?);
- we could have the caller pass an explicit `Span` to `execute`, so they couldn't forget to instrument the future (but if they didn't want to instrument it, they'd have to make something up).

Doing #2 by moving the `.instrument` call to `rsp` rather than the result of `execute` everywhere in the `Service::call` impl seems the cleanest.

Reply:
resolved in https://github.com/penumbra-zone/penumbra/pull/321/commits/8ccb8e13162b939e4d7d2fa524574b39cd977599